core/src/main/java/eu/maveniverse/maven/njord/shared/impl/publisher/basic/SourcesJarValidator.java

```diff
@@ -15,24 +15,23 @@
 import org.eclipse.aether.util.artifact.SubArtifact;
 
 /**
- * Verifies presence of source JAR for every main JAR artifact.
+ * Verifies presence of sources JAR for every main JAR artifact.
  */
-public class SourceJarValidatorFactory extends ValidatorSupport {
-    private static final String JAR = "jar";
-    private static final String SOURCES = "sources";
-
-    public SourceJarValidatorFactory(String name) {
+public class SourcesJarValidator extends ValidatorSupport {
+    public SourcesJarValidator(String name) {
         super(name);
     }
 
     @Override
     public void validate(ArtifactStore artifactStore, Artifact artifact, ValidationContext collector)
             throws IOException {
-        if (artifact.getClassifier().isEmpty() && JAR.equals(artifact.getExtension())) {
+        if (mainJar(artifact)) {
             if (artifactStore.artifactPresent(new SubArtifact(artifact, SOURCES, JAR))) {
                 collector.addInfo("PRESENT");
             } else {
-                collector.addError("MISSING");
+                if (jarContainsJavaClasses(artifact)) {
```

Comment:
In this case, it's a bit more tricky: it should be there if the jar file contains any source files (except for the metadata I suppose but not sure how they identify them - see the example I posted).

Reply:
As in any other case, how to distinguish these two cases:
* main jar classes are generated, not compiled from sources
* user just forgot to attach sources

I just want to keep this simple, would not go into jungle of figuring out "what the build does" by inspecting POM etc

Reply:
My understanding is this:
- Javadoc is simple as as soon as you have a `.class`, you have the javadoc jar
- Sources is a bit more complex. If you look at https://repo1.maven.org/maven2/io/quarkus/quarkus-apicurio-registry-json-schema/3.23.1/, we don't have any `.class` file but... we have a YAML file and thus a source jar containing the YAML file. Funnily enough, there is also a .properties file but it's not included in the source jar. Now I have no idea if Central Portal would be fine if I wasn't providing a source jar for this artifact but it's there when building. Maybe we should have a look at what the `maven-source-plugin` includes in the source jar by default?